### PR TITLE
#236 - incremental update for new repo

### LINKED ISSUE #236: Incremental update fails on new repo
Incremental update does not work on repository without existing metadata. There are two problems:
- the simple one: `Rpm#meta` fails if metadata file was not found
- the hard one: incremental update parses rpms one by one (if parsing is required) when usual update does it simultaneously. Parallelism is invoked on `Rpm` class level. 

As far as I see, solutions can differ depending on our strategy: are we going to keep 2 updates (incremental and simple) or our goal is to leave incremental update only?

### PR DESCRIPTION
**Pull request by olenagerasimova**
Closes #236: I've made incremental update work on new repository. 
Note that `ModifiableMetadata` will be changed further in #314.

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#328](https://github.com/artipie/rpm-adapter/issues/328) is assigned to @vatavuk[/z](https://www.0crat.com/u/vatavuk), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/10dbb3dd-a7a0-4fcd-9e03-e583ff8b3b33); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/main/java/com/artipie/rpm/pkg/ModifiableMetadata.java [hunk 1]
```diff
@@ -46,32 +48,41 @@
...
     }
 
     @Override
     public void brush(final List<String> pkgs) throws IOException {
-        new XmlMetaJoin(this.origin.output().tag())
-            .merge(this.origin.output().file(), this.old);
+        final Optional<Path> old = this.preceding.findAndUnzip();
+        if (old.isPresent()) {
+            new XmlMetaJoin(this.origin.output().tag())
+                .merge(this.origin.output().file(), old.get());
```
Comment (Vatavuk) [inline src/main/java/com/artipie/rpm/pkg/ModifiableMetadata.java:74]: @olenagerasimova extract `old.get()` to a variable to avoid calling it twice

Comment (olenagerasimova) [reply src/main/java/com/artipie/rpm/pkg/ModifiableMetadata.java:74]: @Vatavuk I've removed line to delete the file as files creation and cleaning should be handled on the upper level and should not, in my opinion, be this class responsibility. This file is created in the temp directory created to update rpm repo and will be removed when update is finished along with the directory.

**Comment by 0crat** [comment]
Job was finished in 6 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @vatavuk[/z](https://www.0crat.com/u/vatavuk), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@Vatavuk please make sure you will find at least 3 issues during next CR, thanks

**Comment by Vatavuk** [comment]
@sereshqua ok

**Comment by sereshqua** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
> @0crat quality acceptable  ([here](https://github.com/artipie/rpm-adapter/issues/328#issuecomment-655908989))

@sereshqua There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/farm/issues):

```
PID: 4@a92473af-ed65-4a78-89fa-9bf61fdf677d, thread: PQ-CT2E6TK9B
com.zerocracy.farm.reactive.StkRuntime[76] java.lang.IllegalStateException: org.codehaus.groovy.runtime.InvokerInvocationException in com.zerocracy.stk.pm.cost.pay_cash
org.codehaus.groovy.reflection.CachedMethod[105] org.codehaus.groovy.runtime.InvokerInvocationException: java.io.IOException: Failed to add transaction
com.zerocracy.pm.cost.Ledger[133] java.io.IOException: Failed to add transaction
org.postgresql.core.v3.ConnectionFactoryImpl[245] org.postgresql.util.PSQLException: Connection to pgsql.0dmx.com:5432 refused. Check that the hostname and port are correct and that the postmaster is accepting TCP/IP connections.
java.net.PlainSocketImpl[-2] java.net.ConnectException: Connection refused (Connection refused)
```

[0.53.15](https://github.com/zerocracy/farm): CID: [dafe48fc-e6a4-44d0-be65-994183d0ffc5](https://www.0crat.com/footprint/CT2E6TK9B/dafe48fc-e6a4-44d0-be65-994183d0ffc5), Type: "Pay cash"

**Comment by 0crat** [comment]
> @0crat quality acceptable  ([here](https://github.com/artipie/rpm-adapter/issues/328#issuecomment-655908989))

@sereshqua There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/farm/issues):

```
PID: 4@a92473af-ed65-4a78-89fa-9bf61fdf677d, thread: PQ-CT2E6TK9B
com.zerocracy.farm.reactive.StkRuntime[76] java.lang.IllegalStateException: org.codehaus.groovy.runtime.InvokerInvocationException in com.zerocracy.stk.pm.cost.pay_cash
org.codehaus.groovy.reflection.CachedMethod[105] org.codehaus.groovy.runtime.InvokerInvocationException: java.io.IOException: Failed to add transaction
com.zerocracy.pm.cost.Ledger[133] java.io.IOException: Failed to add transaction
org.postgresql.jdbc.PgConnection[767] org.postgresql.util.PSQLException: This connection has been closed.
```

[0.53.15](https://github.com/zerocracy/farm): CID: [d4419fad-e9c8-4320-8f4a-600a50b11d1f](https://www.0crat.com/footprint/CT2E6TK9B/d4419fad-e9c8-4320-8f4a-600a50b11d1f), Type: "Pay cash"

**Pull request merged**